### ISSUE TITLE
Converted epub sourced from HTML causes KA1 to crash

### ISSUE THREAD

**Issue by da-moo** [geek1011/kepubify#15]
Hi there. I have a bit of an edge case, so I understand if this isn't something you support or want to look into. 

I saved a plain HTML webpage from Firefox (The Org Mode Manual, located at https://orgmode.org/org.html), and then converted that single HTML page to an epub using pandoc. I then used kepubify to convert the epub to a kepub. After loading the book onto my Kobo Aura One, I cannot open the book without the entire device crashing and causing a reboot. 

The content/source material is open source and should be fine to either upload to this issue, or you can easily reproduce the steps yourself. Let me know if you'd like to pursue this issue and I can provide the files I used for the conversation at each step.

**Comment by geek1011** [comment]
Thanks for reporting this issue! I'll be happy to look at it.

May I see the crash logs from the .kobo folder (if any), the source epub, and the converted epub?

**Comment by da-moo** [comment]
I couldn't find anything in the .kobo folder that resembled a crash log, but I attached the source epub and the kepub generated by kepubify.

[org_epubs.zip](https://github.com/geek1011/kepubify/files/1657788/org_epubs.zip)

**Comment by geek1011** [comment]
Does the original epub crash the kobo?

**Comment by da-moo** [comment]
I just tested, and it the original epub works fine. I did notice while my device was connected that the kepub version of the book had its extension mangled to `.epub` instead of the proper `.kepub.epub`. I tested the kepub again after correcting the extension, and the book loads but it slows the device down to where it is almost unusable. I'm guessing that's due to how the source material was structured?

**Comment by da-moo** [comment]
So after some searching around, I discovered that the eBook I was trying to convert (The Org Manual) is sourced in the texinfo format. I then discovered that it is possible to convert from texinfo to epub (and then kepub) via a script/open source toolset. I used this toolset which produced a much better epub than the converted HTML page from pandoc, and the resulting kepub from kepubify is leagues better than the original one I uploaded to this issue. 

TL;DR - Found a much better source for converting to epub than kepub, I no longer have the issue of slowdown/crashes.

**Comment by geek1011** [comment]
Ok, great! Is this issue ready to be closed, or do you want me to look into this more?

**Comment by da-moo** [comment]
That's entirely up to you. I think the original conversion process I used was a bit of an edge case, and I wouldn't expect most users to run into that. Also I'm not sure if kepubify's conversion of the book was the issue, or if the source material was just incredibly inefficient for the Kobo hardware. I'm okay with you closing the issue if that's what you'd like to do.

**Comment by geek1011** [comment]
OK. I had a look at the book files. The reason why you had the problem is because pandoc converts the whole document into one file, instead of splitting into chapters at the titles. The resulting file was 3.5 MB (!) of html. The kobo renderer cannot handle such large files, so it crashed.

Thanks for reporting this issue!

**Comment by geek1011** [comment]
I have reported this issue: kobolabs/Kobo-Reader#68

**Comment by geek1011** [comment]
This should be fixed in the latest firmware (http://geek1011.github.io/KoboStuff/kobofirmware.html). You'll probably need to sideload it or wait a bit, as it has only been released to some of the affiliates so far.
